src/main/java/dev/blaauwendraad/masker/json/KeyMatcher.java

```diff
@@ -217,12 +217,12 @@ private void insert(PreInitTrieNode node, String word, boolean negativeMatch) {
             // check JSONPath first, as it's more specific
             // if found - mask with this config
             // if not found - do not mask
-            if (node != null && node.endOfWord && !node.negativeMatch) {
```

Comment:
Removed `!node.negativeMatch` condition because there can be no negative matched in mask mode, only in allow mode.